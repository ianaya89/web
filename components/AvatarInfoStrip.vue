<template>
  <div class="avatar-strip-info">
    <avatar class="px-2 my-auto" v-bind="$props" :classes="classes" />
    <div class="px-2 flex flex-col">
      <div class="my-auto">
        <div class="font-medium leading-tight pt-1">
          {{ title }}
        </div>
        <div class="text-grey leading-tight">
          {{ info }}
        </div>
      </div>
    </div>
  </div>
</template>


<script>
import Avatar from '@/components/Avatar'

export default {
  name: 'AvatarInfoStrip',

  components: {
    Avatar
  },

  props: {
    alt: {
      type: String,
      default: 'Image alt'
    },
    src: {
      type: String,
      default:
        'https://worldsportlogos.com/wp-content/uploads/2018/01/Chelsea-logo-768x432.png'
    },
    title: {
      type: String,
      default: 'Title'
    },
    info: {
      type: String,
      default: 'Info'
    },
    classes: {
      type: String,
      default: 'w-12 h-12'
    }
  }
}
</script>

<style lang="postcss" scoped>
.avatar-strip-info {
  @apply .w-full .flex .py-2;
  border-bottom: 1px solid rgba(0, 0, 0, 0.2);
  border-left: 4px solid transparent;
  transition: border-left-color 0.15s ease-in-out;
  will-change: border-left-color;

  &:hover {
    border-left-color: #ff553f;
  }
}
</style>
