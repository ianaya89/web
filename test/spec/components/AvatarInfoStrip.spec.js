import { shallowMount } from '@vue/test-utils'
import AvatarInfoStrip from '@/components/AvatarInfoStrip'

describe('AvatarInfoStrip', () => {
  it('renders default', () => {
    const wrapper = shallowMount(AvatarInfoStrip)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders some prop', () => {
    const wrapper = shallowMount(AvatarInfoStrip, {
      propsData: {
        title: 'Custom title',
        info: 'Some basic long lorem ipsum',
        alt: 'My alt'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
