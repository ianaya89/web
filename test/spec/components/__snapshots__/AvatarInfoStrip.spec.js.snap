// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarInfoStrip renders default 1`] = `
<div
  class="avatar-strip-info"
>
  <avatar-stub
    alt="Image alt"
    avatarclasses=""
    class="px-2 my-auto"
    classes="w-12 h-12"
    info="Info"
    rounded="true"
    src="https://worldsportlogos.com/wp-content/uploads/2018/01/Chelsea-logo-768x432.png"
    title="Title"
    tooltipcolor="light"
    tooltipposition="top"
  />
   
  <div
    class="px-2 flex flex-col"
  >
    <div
      class="my-auto"
    >
      <div
        class="font-medium leading-tight pt-1"
      >
        
        Title
      
      </div>
       
      <div
        class="text-grey leading-tight"
      >
        
        Info
      
      </div>
    </div>
  </div>
</div>
`;

exports[`AvatarInfoStrip renders some prop 1`] = `
<div
  class="avatar-strip-info"
>
  <avatar-stub
    alt="My alt"
    avatarclasses=""
    class="px-2 my-auto"
    classes="w-12 h-12"
    info="Some basic long lorem ipsum"
    rounded="true"
    src="https://worldsportlogos.com/wp-content/uploads/2018/01/Chelsea-logo-768x432.png"
    title="Custom title"
    tooltipcolor="light"
    tooltipposition="top"
  />
   
  <div
    class="px-2 flex flex-col"
  >
    <div
      class="my-auto"
    >
      <div
        class="font-medium leading-tight pt-1"
      >
        
        Custom title
      
      </div>
       
      <div
        class="text-grey leading-tight"
      >
        
        Some basic long lorem ipsum
      
      </div>
    </div>
  </div>
</div>
`;
